app/src/auth.ts

```diff
@@ -94,6 +94,9 @@ export async function refresh({ navigate }: LogoutOptions = { navigate: true }):
 	// Allow refresh during initial page load, skip if not logged in
 	if (!firstRefresh && !appStore.authenticated) return;
 
+	// Skip access token refreshing if it is still fresh
+	if (appStore.accessTokenExpiry && Date.now() < appStore.accessTokenExpiry - SDK_AUTH_REFRESH_BEFORE_EXPIRES) return;
```

Comment:
```suggestion
	if (appStore.accessTokenExpiry && Date.now() < appStore.accessTokenExpiry - SDK_AUTH_REFRESH_BEFORE_EXPIRES) {
		try {
			await sdk.request(readMe());
		} catch {
			await logout({ navigate, reason: LogoutReason.SESSION_EXPIRED });
		}

		return;
	}
```

Reply:
👍 Shall we fetch the `id` field with `sdk.request(readMe({ fields: ['id'] }))`, reducing the response body size?

Reply:
Good suggestion, should we keep that "SESSION EXPIRED" reason?

Reply:
Yes, I believe so, and it's consistent with the existing logic.
Changes were pushed in https://github.com/directus/directus/pull/22464/commits/dfa4a078ea9e3d33466e89074087f7e39bf3d2de where the session validation is added into the existing try-catch-finally clause.